sdk/rust/oak_tests/src/lib.rs

```diff
@@ -677,6 +697,76 @@ pub unsafe extern "C" fn channel_find(buf: *const u8, size: usize) -> u64 {
     handle
 }
 
+/// Test framework implementation of dynamic Node creation.
+#[no_mangle]
+pub unsafe fn node_create(buf: *const u8, len: usize, handle: u64) -> u32 {
+    let name = node_name();
+    debug!("{{{}}}: node_create({:?}, {})", name, buf, len);
+    let mut data = Vec::with_capacity(len as usize);
+    std::ptr::copy_nonoverlapping(buf, data.as_mut_ptr(), len as usize);
+    data.set_len(len as usize);
+    let contents = String::from_utf8(data).unwrap();
+    debug!("{{{}}}: node_create('{}', {})", name, contents, handle);
+
+    // First, find the code referred to by the contents name.
+    let entrypoint = match RUNTIME.read().unwrap().entrypoint(&contents) {
```

Comment:
Is it worth just getting the `RUNTIME.write()` lock once and for all for the lifetime of this method?

Reply:
Makes sense; I've added an `OakRuntime::node_create` method to do most of the work (just leaving the thread creation and join-handle-registration to be done afterwards here).